### ISSUE TITLE
Make `parse_slice()` const?

### ISSUE THREAD

**Issue by mlund** [embedded-graphics/tinytga#20]
- Version of tinytga in use (if applicable): 0.5.0

## Description of the problem/feature request/other

I'm experimenting with embedded-graphics on a very restricted system with only 40 kB available RAM, and while it does work with primitives like circles, fonts etc., using tinytga, the size grows up to 80 kB bytes or so while loading a 2 kB file. I therefore wonder if `fn parse_slice()` could be made `const` so that image processing is done at compile-time? I realise that this could be an elaborate task due to the limitations of [constant expressions].

**Comment by rfuest** [comment]
I don't think that will be possible, but I also don't think that this is your problem. I'm pretty sure I've successfully used tinytga on a microcontroller with 8 kB of RAM in the past.

Can you provide a bit more information about the system, like what architecture/controller you are working with? Or can you even make the code that you use to draw the image and an example image available?

**Comment by mlund** [comment]
Thanks for the quick reply. It could be that it's the rust compiler that does something unexpected. The target is a a Commodore 64 8-bit computer using llvm-mos/rust-mos. I'll shortly push the display driver to a crate with instructions how to run this (which involves Docker due to the special toolchain).

**Comment by rfuest** [comment]
Now that's an interesting project, I'm happy to help with that. I had never expected to use any Rust code I've written to run on my C64, but I'll definitely try it when we get this working.

**Comment by mlund** [comment]
Sounds great - here's preview of what I have so far ;-)

https://github.com/embedded-graphics/tinytga/assets/3248398/07f29486-9954-4d07-ada0-2f316ab70aa8

**Comment by mlund** [comment]
The `retro-display` display driver is out: https://crates.io/crates/retro-display

**Comment by mlund** [comment]
This is great, and 2K is not bad at all. Yes, I started looking into conversion from RGB by calculating the color difference and realised that this can be done in many ways. There are crates available, but I didn't continue investigating.
